From c9f84ef9b84d81ff4383b80372902f307e97d73d Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Mon, 4 Aug 2025 14:22:03 +0200
Subject: [PATCH] tests: Fix test failures of Java tests on Cygwin.

* bootstrap.conf (gnulib_modules): Add nlcanon.
* tests/local.at (AT_JAVA_PARSER_CHECK): Add an nlcanon.sh invocation.
---
 bootstrap.conf | 1 +
 tests/local.at | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/bootstrap.conf b/bootstrap.conf
index 2e9513c3..8a924973 100644
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -35,6 +35,7 @@ gnulib_modules='
   linked-list
   malloc-gnu
   mbfile mbswidth
+  nlcanon
   obstack
   obstack-printf
   perror progname
diff --git a/tests/local.at b/tests/local.at
index ee542f9e..2bbaeb98 100644
--- a/tests/local.at
+++ b/tests/local.at
@@ -1599,7 +1599,8 @@ AT_CHECK([sed >&2 -e '/^profiling:.*:Merge mismatch for summaries/d' stderr],
 # AT_JAVA_PARSER_CHECK(COMMAND, EXIT-STATUS, EXPOUT, EXPERR, [PRE])
 # -----------------------------------------------------------------
 m4_define([AT_JAVA_PARSER_CHECK],
-[AT_CHECK([$5[ $SHELL ../../../javaexec.sh ]$1], [$2], [$3], [$4])])
+[AT_CHECK([$5[ $SHELL ../../../nlcanon.sh windows-based stdout,stderr $SHELL ../../../javaexec.sh ]$1],
+          [$2], [$3], [$4])])
 
 
 # AT_PARSER_CHECK(COMMAND, EXIT-STATUS, EXPOUT, EXPERR, [PRE])
-- 
2.50.1

